windows/win_nssm.ps1

```diff
@@ -0,0 +1,547 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2015, George Frank <george@georgefrank.net>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+$ErrorActionPreference = "Stop"
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$params = Parse-Args $args;
+$result = New-Object PSObject;
+Set-Attr $result "changed" $false;
+
+
+If ($params.name)
+{
+    $name = $params.name
+}
+Else
+{
+    Fail-Json $result "missing required argument: name"
+}
+
+If ($params.state)
+{
+    $state = $params.state.ToString().ToLower()
+    $validStates = "present", "absent", "started", "stopped", "restarted"
+    
+    # These don't really fit the declarative style of ansible
+    # If you need to do these things, you can just write a command for it
+    # "paused", "continued", "rotated"
+
+    If ($validStates -notcontains $state)
+    {
+        Fail-Json $result "state is $state; must be one of: $validStates"
+    }
+}
+else
+{
+    $state = "present"
+}
+
+If ($params.application)
+{
+    $application = $params.application
+}
+Else
+{
+    $application = $null
+}
+
+If ($params.app_parameters)
+{
+    $appParameters = $params.app_parameters
+}
+Else
+{
+    $appParameters = $null
+}
+
+If ($params.stdout_file)
+{
+    $stdoutFile = $params.stdout_file
+}
+Else
+{
+    $stdoutFile = $null
+}
+
+If ($params.stderr_file)
+{
+    $stderrFile = $params.stderr_file
+}
+Else
+{
+    $stderrFile = $null
+}
+
+Function Service-Exists
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name
+    )
+
+    return ,[bool](Get-Service "$name" -ErrorAction SilentlyContinue)
+}
+
+Function Nssm-Remove
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name
+    )
+
+    if (Service-Exists -name $name)
+    {
+        $cmd = "nssm stop ""$name"""
+        $results = invoke-expression $cmd
```

Comment:
In the statement below you check and handle an error code, but not here. I
don't know nssm's behaviour. Is it not necessary to check the exit code
here?

Sorry for not being clear.

--
Sent from my phone. Please excuse typos and brevity, but never text speak.
On 2 Sep 2015 16:11, "georgefrank" <notifications@github.com> wrote:

> In windows/win_nssm.ps1
> <https://github.com/ansible/ansible-modules-extras/pull/695#discussion_r38543663>
> :
>
> > +
> > +    return ,[bool](Get-Service "$name" -ErrorAction SilentlyContinue)
> > +}
> > +
> > +Function Nssm-Remove
> > +{
> > +    [CmdletBinding()]
> > +    param(
> > +        [Parameter(Mandatory=$true)]
> > +        [string]$name
> > +    )
> > +
> > +    if (Service-Exists -name $name)
> > +    {
> > +        $cmd = "nssm stop ""$name"""
> > +        $results = invoke-expression $cmd
>
> Can you elaborate on what you're asking here? I'm not sure I understand
> the intent of the question.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible-modules-extras/pull/695/files#r38543663>
> .
>